docs/root/configuration/http/http_filters/rate_limit_quota_filter.rst

```diff
@@ -0,0 +1,168 @@
+* :ref:`v3 API reference <envoy_v3_api_msg_extensions.filters.http.rate_limit_quota.v3.RateLimitQuotaFilterConfig>`
+
+This filter provides implementation of the global rate limit quota :ref:`protocol <envoy_v3_api_file_envoy/service/rate_limit_quota/v3/rlqs.proto>`.
+The rate limit quota service (RLQS) provides quota assignments to each Envoy instance connected to the RLQS service. In addition to enforcing rate limit quota assignments,
+this filter periodically reports request rates to the RLQS, allowing the RLQS server to rebalance quota assignments between Envoy instances based on the real-time
+individual load of each Envoy instance. When quota assignments change, the RLQS proactively pushes the new assignment to Envoy.
+
+The HTTP rate limit quota filter will call the rate limit quota service when it is configured in the HTTP connection manager filter chain. Filter configuration
+defines the RLQS service and definitions of quota buckets that will receive quota assignments from server. Qouta buckets are defined by a set of matchers that determine
+if a request is subject to the rate limit quota assigned to that bucket. Each matcher can contain multiple buckets by the means of the
+:ref:`bucket_id_builder <envoy_v3_api_field_extensions.filters.http.rate_limit_quota.v3.RateLimitQuotaBucketSettings.bucket_id_builder>`. The bucket ID builder allows
+quota buckets to be generated either dynamically based on request attributes, such as request header value or statically based on the configuration.
+
+If a request does not match any set of matchers then quota assignment for the "catch all" bucket configured by the ``on_no_match`` field of the
+:ref:`bucket_matchers <envoy_v3_api_field_extensions.filters.http.rate_limit_quota.v3.RateLimitQuotaFilterConfig.bucket_matchers>` is applied. If the ``on_no_match``
+configuration is not provided, all unmatched requests are not rate limited (i.e. fail-open).
+
+Bucket definitions can be overridden by the virtual host or route configurations. The more specific definition completely overrides the less specific definition.
+
+Initially all Envoy's quota assignments are empty. The rate limit quota filter requests quota assignment from RLQS when the request matches to a bucket for the first time.
+The behavior of the filter while it waits for the initial assignment is determined by the ``no_assignment_behavior`` value. In this state, requests can either all be
+immediately allowed, denied or enqueued until quota assignment is received.
+
+A quota assignment may have associated :ref:`time to live <envoy_v3_api_field_service.rate_limit_quota.v3.RateLimitQuotaResponse.BucketAction.QuotaAssignmentAction.assignment_time_to_live>`.
+The RLQS is expected to update the assignment before TTL runs out. If RLQS failed to update the assignment and its TTL
+has expired, the filter can be configured to continue using the last quota assignment or fall back to a value predefined in the
```

Comment:
Does using the last quota assignment imply that there's actually no TTL? Or is TTL just used to define the amount of time to request new quota from the RLQS server?

Reply:
Using last quota assignment is one of two options when TTL expires and no new assignment from server has sent back. 
This assignment_ttl represents the lifetime of assignment but is not related to the interval that RLQS server updates the assignment. If server doesn't update the assignment before it get expired, we will use `expired_assignment_behavior` to handle this case.